uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeavePeriodEntitlementTest.php

```diff
@@ -106,4 +110,251 @@ public function testGetBreakdownWhenOnlyEntitlementIdIsPassed() {
+   */
+  public function testGetEntitlementDoesNotAcceptPeriodIdWithoutContactId() {
+    civicrm_api3('LeavePeriodEntitlement', 'getEntitlement', ['period_id' => 1]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage You must include either the id of a specific entitlement, or both the contact and period id
+   */
+  public function testGetEntitlementWhenAllParametersArePassed() {
+    civicrm_api3('LeavePeriodEntitlement', 'getEntitlement', ['entitlement_id'=> 1, 'period_id' => 1, 'contact_id'=>1]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage You must include either the id of a specific entitlement, or both the contact and period id
+   */
+  public function testGetEntitlementWhenEntitlementIdAndPeriodIdArePassed() {
+    civicrm_api3('LeavePeriodEntitlement', 'getEntitlement', ['entitlement_id'=> 1, 'contact_id'=>1]);
+  }
+
+  public function testGetEntitlementCanReturnTheEntitlementsForAContactInAPeriod() {
+    $contact = ContactFabricator::fabricate();
+
+    $type1 = AbsenceTypeFabricator::fabricate();
+    $type2 = AbsenceTypeFabricator::fabricate();
+    $type3 = AbsenceTypeFabricator::fabricate();
+
+    $period1 = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'end_date' => CRM_Utils_Date::processDate('2016-12-31')
+    ]);
+
+    $period2 = AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2017-01-01'),
+      'end_date' => CRM_Utils_Date::processDate('2017-12-31')
+    ]);
+
+    $periodEntitlement1 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'type_id' => $type1->id,
+      'period_id' => $period1->id
+    ]);
+
+    $periodEntitlement2 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'type_id' => $type2->id,
+      'period_id' => $period1->id
+    ]);
+
+    $periodEntitlement3 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'type_id' => $type2->id,
+      'period_id' => $period2->id
+    ]);
+
+    $periodEntitlement4 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'type_id' => $type3->id,
+      'period_id' => $period2->id
+    ]);
+
+    $result = civicrm_api3('LeavePeriodEntitlement', 'getEntitlement', [
+      'contact_id' => $contact['id'],
+      'period_id' => $period1->id
+    ]);
+
+    //For the period 1, the contact only has entitlements for Types 1 and 2
+    $expected = [
+      'is_error' => 0,
+      'version' => 3,
+      'count' => 2,
+      'values' => [
+        [
+          'id' => $periodEntitlement1->id,
+          'entitlement' => 0,
+        ],
+        [
+          'id' => $periodEntitlement2->id,
+          'entitlement' => 0,
+        ]
+      ]
+    ];
+
+    $this->assertEquals($expected, $result);
+
```

Comment:
I know I've said this was a good idea sometime ago, but I've been thinking about it and I changed my mind. The problem with `assertArraySubset` is that it will also not fail if the returned values change. For example, if we change this:

```php
[
  'values' => [
    'id' => 1,
    'contact_id'  => 1
  ]
]
```

To this:

```php
[
  'values' => [
    'id' => 1,
    'contact_id'  => 1,
    'blablabla' => 'blablabla', 
  ]
]
```

Testing only for a subset would not make the test fail, even though the response has a new field now.